end_to_end/eval_assert.py

```diff
@@ -20,13 +21,32 @@ def read(metrics_file, target):
   return avg
 
 
+def test_tflops(metrics_file, target, threshold):
+  """Asserts over tflops values"""
+  avg_tflops = read(metrics_file, target)
+  assert avg_tflops >= threshold
+
+
+def test_checkpointing(metrics_file, target):
+  """Asserts over loss values from loaded checkpoint"""
+  metrics_file_saved = 'saved_' + metrics_file
+  metrics_file_restored = 'restored_' + metrics_file
+
+  with open(metrics_file_saved, 'r', encoding='utf8') as saved,\
+    open(metrics_file_restored, 'r', encoding='utf8') as restored:
+    saved_loss = json.loads(saved.readlines()[-1])[target]
+    restored_loss = json.loads(restored.readlines()[0])[target]
+    assert isclose(saved_loss, restored_loss, rel_tol=0.5)
+
+
 def main(argv: Sequence[str]) -> None:
 
   _, metrics_file, threshold, target = argv
 
-  avg_val = read(metrics_file, target)
-
-  assert avg_val >= float(threshold)
+  if target == 'perf/per_device_tflops_per_sec':
+    test_tflops(metrics_file, target, float(threshold))
+  elif target == 'learning/loss':
+    test_checkpointing(metrics_file, target)
```

Comment:
I think future tests may use/re-use different targets. Can we instead add a new arg named something like "test_scenario" (example inputs "checkpointing", "performance", etc) to the main of eval_assert which tells us which tests to run?